### ISSUE TITLE
Server restart

### ISSUE THREAD

**Issue by NarsunProjects** [agenda/agenda#573]
Even there are many issues for the same question but none worked for me! When I restart the server how to catch up all the job that were one time scheduled using 
agenda.schedule(date, 'name');
I tried many ways but not getting my jobs running if there were 5 mins in running the job. It skips it and listen to new jobs only! All I want that on restart server, agenda.start() also runs the jobs that are yet to be run!

**Comment by malas** [comment]
it seems like you should use a job queue instead of cronjob like mechanism which agenda is.

cronjobs are used to run tasks periodically, but **not** to ensure that every one of them is completed in case the server is down.

**Comment by bonesoul** [comment]
bull is way to go then?

**Comment by sam-gikian** [comment]
Is there any solution to this? 

I have tried all the solutions discussed on Git but Agenda doesn't run the jobs scheduled before server restart. I've tried graceful exit and even updating the docs in MongoDB with new time. And my lockedAt is also null which means the job is not locked. 

p.s: I'm on the latest version, Agenda 2.0.2

**Comment by rajavijayavendan** [comment]
@NarsunProjects Hi,
Do you have any solution for running already scheduled jobs after restart?

**Comment by NarsunProjects** [comment]
Yeah I do. I've done that through customization in Agenda. Added in schema where I store the whole job as a text in mongo. When the server restarts, I've defined the agenda function that schedules already running jobs again with the job stored as a text.
